arch/arm/core/aarch64/isr_wrapper.S

```diff
@@ -83,15 +83,28 @@ SECTION_FUNC(TEXT, _isr_wrapper)
 	cmp	x2, x3
 	beq	exit
 
+	/*
+	 * As expected by z_arm64_context_switch():
+	 *
+	 * x16 <- new_thread->switch_handle
+	 * x17 <- old_thread->switch_handle
+	 */
+	mov	x16, x3
+	mov	x17, x2
+
+	/* Now we can fetch the thread to run from the ready queue cache */
+	ldr	x2, [x1, #_kernel_offset_to_ready_q_cache]
+	str	x2, [x1, #_kernel_offset_to_current]
```

Comment:
Since `ready_q_cache` is already in `x16`, we could just do:
```
	str	x16, [x1, #_kernel_offset_to_current]
```

Reply:
Design-wise, this isn't really where you want to be.  You can't use _current at all in SMP (because you'd have to recompute it on the fly every time a CPU selected it to run, which defeats the point of the optimization).  The intent is that arch_switch() architectures call z_get_next_switch_handle() out of their interrupt exit path, and rely on the built-in z_swap() in the kernel which does the scheduler logic for you and just provides the switch handle and an output slot for the saved one.

Reply:
So IIUC since I'm using the thread pointer as `switch_handler` (see https://github.com/zephyrproject-rtos/zephyr/pull/22877/files#diff-f51234a0c31a6e7a874cac5d2595b0e9R84) I cannot pass the `_current` pointer directly to `z_get_next_switch_handle()`in the interrupt exit path?

Reply:
@andyross ^ as for my question above, is it allowed to directly pass `_current` to `z_get_next_switch_handle()`?

Reply:
Pushed a new revision